commands/bmc

```diff
@@ -0,0 +1,252 @@
+#!/bin/bash -eu
+
+#
+# CLI: General BMC configuration
+#
+
+# Format date in a non-standard way
+function format_date {
+  local compact="$1"
+  local utc="$2"
+  local sec="$3"
+
+  if [[ -n "${compact}" ]] && [[ -n "${utc}" ]]; then
+    date --date "@${sec}" --utc +'%F %T UTC'
+  elif [[ -n "${compact}" ]]; then
+    date --date "@${sec}" +'%F %T UTC%z'
+  elif [[ -n "${utc}" ]]; then
+    date --date "@${sec}" --utc +'%a %d %b %Y %T UTC'
+  else
+    date --date "@${sec}" +'%a %d %b %Y %T UTC%z'
+  fi
+}
+
+# @doc cmd_info
+# General BMC information
+function cmd_info {
+  subcommand "$@"
+}
+
+# @doc cmd_info_uptime
+# Show BMC uptime
+#   -c, --compact   Show last boot time in compact format
+#   -u, --utc       Show last boot time in UTC timezone
+function cmd_info_uptime {
+  local compact=""
+  local utc=""
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      -c | --compact) compact="y";;
+      -u | --utc) utc="y";;
+      *) abort_badarg "$1";;
+    esac
+    shift
+  done
+
+  local sec="$(awk -F. '{print $1}' /proc/uptime)"
+  local up="$(($(date +%s) - sec))"
+
+  printf 'Uptime is %dd %dh %dm %ds.\n' $((sec / 86400)) \
+                                        $((sec % 86400 / 3600)) \
+                                        $((sec % 3600 / 60)) \
+                                        $((sec % 60))
+  echo -n "Last BMC start was on "
+  format_date "${compact}" "${utc}" "${up}"
+}
+
+# @sudo cmd_info_version admin,operator,user
+# @doc cmd_info_version
+# Show version of BMC and BIOS/UEFI
+#   -b, --bmc   Show BMC version
+#   -u, --bios  Show BIOS/UEFI version
+function cmd_info_version {
+  local bmc=""
+  local bios=""
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+      -b | --bmc) bmc="y";;
+      -u | --bios) bios="y";;
+      *) abort_badarg "$1";;
+    esac
+    shift
+  done
+
+  if [[ -z "${bmc}${bios}" ]]; then
+    bmc="y"
+    bios="y"
+  fi
+
+  if [[ -n "${bmc}" ]]; then
+    fwupdate --version | grep "^BMC "
+  fi
+  if [[ -n "${bios}" ]]; then
+    fwupdate --version | grep "^Host "
+  fi
+}
+
+# @doc cmd_datetime
+# BMC date and time configuration
+function cmd_datetime {
+  subcommand "$@"
+}
+
+# @sudo cmd_datetime_set admin
+# @doc cmd_datetime_set
+# Set system time manually
+#   YYYY-MM-DD-HH-mm-SS   Time to set
+function cmd_datetime_set {
+  if [[ $# -eq 0 ]] || [[ $1 =~ ^\d{4}-\d\d-\d\d-\d\d-\d\d-\d\d$ ]]; then
```

Comment:
Hm... Looks strange. Try to run it several times with the different values ;)
Maybe it would better to use here an ISO format ("YYYY-MM-DD HH:mm:SS")?
Or it might be just 14 digits with any combinations of [` `, `-`, `:`] as delimiters or without ones.
It seems to me, there should be either well-known format or any possible combinations here.

Reply:
I can't, this format is required by our internal doc (see VEGMAN CLI Proposal).

Reply:
I will do as soon as it will be described in the proposal.